### PR TITLE
Add support for HDR static Metadata

### PR DESCRIPTION
**Pull request by peak3d**
## Description
This PR implements support for HDR static Metadata by adding a bunch of color fields to DVDStreamInfo struct which allows setting up decoder / renderer for HDR playback.

Added:
- AVMasteringDisplayMetadata
- AVContentLightMetadata
- AVColorRange
- AVColorSpace
- AVColorTransferCharacteristic
- AVColorPrimaries

For android the implementation is part of this PR in the second commit

## Motivation and Context
Play VP9 profile2 HDR Youtube content
Note, that for HDR playback inputstream.adaptive > 2.3.22 has to be used.

## How Has This Been Tested?
- Youtube addon on AFTV / Android
- Enable VP9 and HDR in plugin.video.youtube addon
- Search streams by "vp9 hdr"
- Playback

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Notes
- Inputstream api version has changed, but not Min_Version -> There should be no compatibility issues with older inputstream based binary addons
- Even this PR includes the full Android implementation for static HDR metadata, it has no effect on NVIDIA shield because of missing VP9 profile 2 hardware support. It works well on e.g. AFTV devices.
- Android SDK API >=24 required to see the results

### CODE REVIEW

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/StreamCodec.h
Comment (Rechi) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/StreamCodec.h]: add a comma at the end, so the line doesn't have to be change if an additional value is added later

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h
Comment (Rechi) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h]: move curly bracket into new line, https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#31-braces

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h
Comment (Rechi) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h]: remove the vertical alignment in the whole block, https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#35-vertical-alignment

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h
Comment (Rechi) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h]: remove the vertical alignment, https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#35-vertical-alignment

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h
Comment (Rechi) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h]: remove the vertical alignment, https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#35-vertical-alignment

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h
Comment (Rechi) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h]: remove the vertical alignment, https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#35-vertical-alignment

**Review by Rechi — COMMENTED** [review]
Please change the PR title to something more descriptive

**Review by garbear — APPROVED** [review]
Can I ask for a space after `//` and before the comment?

**Comment by peak3d** [comment]
@garbear @Rechi thx for reviewing, the latest force push includes the code style formatting, pls. look at it again.

**Comment by CiNcH83** [comment]
Is there a test build of this branch? I already recognized that playing VP9.2 from file inside Kodi does not trigger HDR mode on Sony BRAVIA. Google Play Movies and YouTube apps do...

**Comment by peak3d** [comment]
@CiNcH83 There is a build apk (click on "Show all checks", then follow jenkins path) but inputstream.adaptive is not bundled -> you'll have to wait for a nightly after merge or build it by yourown including the latest inputstream.adaptive master or Matrix branch.

**Review by Rechi — COMMENTED** [review]
It's fine for me, but I would have only formatted the code which was actually changed or at least moved the formatting of code not related to this PR into a separate commit.
Now it's hard to spot the real code changes in all those formatting changes.

**Comment by peak3d** [comment]
I have already reset the original changes but will go your suggested way next time @Rechi .

**Comment by CiNcH83** [comment]
Tried latest nightly on my BRAVIA and played VP9.2 from file (WebM and remuxed to MKV; downloaded from YouTube via youtube-dl). The TV still does not switch to HDR mode. HEVC with HDR10 and HLG (inside MP4 and TS containers) properly triggers HDR mode.

VP9.2 works properly for YouTube and Google Play Movies apps.

**Comment by peak3d** [comment]
This initial commit only works with inputstream.adaptive, downloaded / remuxed streams are not yet implemented. If you have issues playing via inputstream.adaptive (DASH / VP9-HDR enabled in youtube addon), pls. provide as always a kodi debug log file.

**Comment by CiNcH83** [comment]
I now installed the YouTube add-on but I am not sure whether Kodi plays VP9.2. HDR mode is not triggered at least. I tried playing [The World in HDR in 4K](https://www.youtube.com/watch?v=tO01J-M3g0U). Only the VP9 streams are enumerated and I can go up to the 2160p60 resolution. Playback however is quite jerky at that resolution.

I am also appending a log of the playback attempt.

[[LOG](http://paste.kodi.tv/navofujuxi)]

**Comment by peak3d** [comment]
Starting Kodi (19.0-ALPHA1 Git:20190827-3a7e2f2b6b). Platform: Android ARM 32-bit
In this version the PR is NOT included

**Comment by CiNcH83** [comment]
Sure? From git history I can see that 3a7e2f2b6b is the first one to include the PR, no?

**Comment by peak3d** [comment]
You are right, version is Ok, I'll add some more logging for HDR in the next step.

But I have the impression that the requested inputstream.adaptive is not picked from out build system:

CDVDVideoCodecAndroidMediaCodec::Open hints: Width 3840 x Height 2160, Fpsrate 60000 / Fpsscale 1001, CodecID 167, Level -99, Profile -99, PTS_invalid 0, Tag 0, Extradata-Size: 0

Pfofile is not set.

**Comment by CiNcH83** [comment]
I now installed today's nightly and inputstream.adaptive version is still 2.3.22.

**Comment by ksooo** [comment]
> I now installed today's nightly and inputstream.adaptive version is still 2.3.22.

@peak3d you need to change repo-binary-addons to point to Matrix branch to pull 2.4.0 into nightly build.

https://github.com/xbmc/repo-binary-addons/blob/Matrix/inputstream.adaptive/inputstream.adaptive.txt point still to Leia (which is 2.3.22).

**Comment by CiNcH83** [comment]
I again tried some VP9.2 via Kodi YouTube add-on on my BRAVIA with latest nightly (which now includes inputstream.adaptive 2.5.0). The BRAVIA now properly switches to HDR.
2160p60 still causes quite some stuttering, also see [here](https://forum.kodi.tv/showthread.php?tid=343267). It might also be due to badly optimized drivers on Sony/MediaTek side though.

VP9.2 from file doesn't cause an HDR mode switch yet. But you already commented on that.

**Pull request merged**
